river/metrics/balanced_accuracy.py

```diff
@@ -0,0 +1,56 @@
+from . import base
+from . import confusion
+
+__all__ = ['BalancedAccuracy']
+
+class BalancedAccuracy(base.BinaryMetric, base.MultiClassMetric):
+    """Balanced Accuracy, which is the average of recall obtained on each class, 
+    is used to deal with imbalanced datasets in binary and multiclass classification problems.
+    
+    Example:
+    
+        >>> from river import metrics
+        >>> y_true = [True, False, True, True, False, True]
+        >>> y_pred = [True, False, True, True, True, False]
+        
+        >>> metric = metrics.BalancedAccuracy()
+        >>> for yt, yp in zip(y_true, y_pred):
+        ...     metric = metric.update(yt, yp)
+        
+        >>> metric
+        BalancedAccuracy : 62.50%
+        
```

Comment:
Could you also add the example from scikit-learn? Use `0` and `1` instead of `False` and `True`, it's a good unit test to check that the metric works with both types.